src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -686,10 +686,16 @@ ss::future<bool> sharded_store::is_compatible(
         co_return true;
     }
 
-    // Currently support PROTOBUF, AVRO
-    if (
-      new_schema.type() != schema_type::avro
-      && new_schema.type() != schema_type::protobuf) {
+    // Currently support JSON, PROTOBUF, AVRO
+    if (![type = new_schema.type()] {
+            switch (type) {
+            case schema_type::avro:
+            case schema_type::protobuf:
+            case schema_type::json:
+                return true;
+            }
+            return false;
+        }()) {
```

Comment:
@andijcr not to be a curmudgeon, but let's not put multi-line IILs into if statement... it doesn't improve anything, and it's hard to read the code.

Reply:
Should i push a fix?

Reply:
> @andijcr not to be a curmudgeon, but let's not put multi-line IILs into if statement... it doesn't improve anything, and it's hard to read the code.

I think I wrote that in a comment/suggestion, so it's on me. It's a bit awkward, but felt like a reasonable trade-off at the time.